From 170724c7b7e9a648872bbec27fd3eb49dff18473 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Fri, 23 Feb 2024 14:58:23 +0100
Subject: [PATCH 0890/1141] ARM: dts: stm32: rework exti node for upstream on
 stm32mp151

Due to upstream review, drop the interrupt-map child node and use
the interrupts-extended property.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I25c77f87d21413e2f119e1b33f87a1c52ba4af76
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/363288
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm/boot/dts/stm32mp151.dtsi | 134 +++++++++++++++++-------------
 1 file changed, 75 insertions(+), 59 deletions(-)

diff --git a/arch/arm/boot/dts/stm32mp151.dtsi b/arch/arm/boot/dts/stm32mp151.dtsi
index 58139f82deaf..680fe741cfa6 100644
--- a/arch/arm/boot/dts/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/stm32mp151.dtsi
@@ -374,68 +374,84 @@ exti: interrupt-controller@5000d000 {
 			compatible = "st,stm32mp1-exti";
 			interrupt-controller;
 			#interrupt-cells = <2>;
-			#address-cells = <0>;
 			reg = <0x5000d000 0x400>;
 			hwlocks = <&hsem 1 1>;
 			wakeup-parent = <&pwr_irq>;
-
-			exti-interrupt-map {
-				#address-cells = <0>;
-				#interrupt-cells = <2>;
-				interrupt-map-mask = <0xffffffff 0>;
-				interrupt-map =
-					<0  0 &intc GIC_SPI 6   IRQ_TYPE_LEVEL_HIGH>,
-					<1  0 &intc GIC_SPI 7   IRQ_TYPE_LEVEL_HIGH>,
-					<2  0 &intc GIC_SPI 8   IRQ_TYPE_LEVEL_HIGH>,
-					<3  0 &intc GIC_SPI 9   IRQ_TYPE_LEVEL_HIGH>,
-					<4  0 &intc GIC_SPI 10  IRQ_TYPE_LEVEL_HIGH>,
-					<5  0 &intc GIC_SPI 23  IRQ_TYPE_LEVEL_HIGH>,
-					<6  0 &intc GIC_SPI 64  IRQ_TYPE_LEVEL_HIGH>,
-					<7  0 &intc GIC_SPI 65  IRQ_TYPE_LEVEL_HIGH>,
-					<8  0 &intc GIC_SPI 66  IRQ_TYPE_LEVEL_HIGH>,
-					<9  0 &intc GIC_SPI 67  IRQ_TYPE_LEVEL_HIGH>,
-					<10 0 &intc GIC_SPI 40  IRQ_TYPE_LEVEL_HIGH>,
-					<11 0 &intc GIC_SPI 42  IRQ_TYPE_LEVEL_HIGH>,
-					<12 0 &intc GIC_SPI 76  IRQ_TYPE_LEVEL_HIGH>,
-					<13 0 &intc GIC_SPI 77  IRQ_TYPE_LEVEL_HIGH>,
-					<14 0 &intc GIC_SPI 121 IRQ_TYPE_LEVEL_HIGH>,
-					<15 0 &intc GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>,
-					<16 0 &intc GIC_SPI 1   IRQ_TYPE_LEVEL_HIGH>,
-					<19 0 &intc GIC_SPI 3   IRQ_TYPE_LEVEL_HIGH>,
-					<21 0 &intc GIC_SPI 31  IRQ_TYPE_LEVEL_HIGH>,
-					<22 0 &intc GIC_SPI 33  IRQ_TYPE_LEVEL_HIGH>,
-					<23 0 &intc GIC_SPI 72  IRQ_TYPE_LEVEL_HIGH>,
-					<24 0 &intc GIC_SPI 95  IRQ_TYPE_LEVEL_HIGH>,
-					<25 0 &intc GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>,
-					<26 0 &intc GIC_SPI 37  IRQ_TYPE_LEVEL_HIGH>,
-					<27 0 &intc GIC_SPI 38  IRQ_TYPE_LEVEL_HIGH>,
-					<28 0 &intc GIC_SPI 39  IRQ_TYPE_LEVEL_HIGH>,
-					<29 0 &intc GIC_SPI 71  IRQ_TYPE_LEVEL_HIGH>,
-					<30 0 &intc GIC_SPI 52  IRQ_TYPE_LEVEL_HIGH>,
-					<31 0 &intc GIC_SPI 53  IRQ_TYPE_LEVEL_HIGH>,
-					<32 0 &intc GIC_SPI 82  IRQ_TYPE_LEVEL_HIGH>,
-					<33 0 &intc GIC_SPI 83  IRQ_TYPE_LEVEL_HIGH>,
-					<43 0 &intc GIC_SPI 75  IRQ_TYPE_LEVEL_HIGH>,
-					<44 0 &intc GIC_SPI 98  IRQ_TYPE_LEVEL_HIGH>,
-					<47 0 &intc GIC_SPI 93  IRQ_TYPE_LEVEL_HIGH>,
-					<48 0 &intc GIC_SPI 138 IRQ_TYPE_LEVEL_HIGH>,
-					<50 0 &intc GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>,
-					<52 0 &intc GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>,
-					<53 0 &intc GIC_SPI 141 IRQ_TYPE_LEVEL_HIGH>,
-					<54 0 &intc GIC_SPI 135 IRQ_TYPE_LEVEL_HIGH>,
-					<55 0 &pwr_irq 0 IRQ_TYPE_EDGE_FALLING 0>,
-					<56 0 &pwr_irq 1 IRQ_TYPE_EDGE_FALLING 0>,
-					<57 0 &pwr_irq 2 IRQ_TYPE_EDGE_FALLING 0>,
-					<58 0 &pwr_irq 3 IRQ_TYPE_EDGE_FALLING 0>,
-					<59 0 &pwr_irq 4 IRQ_TYPE_EDGE_FALLING 0>,
-					<60 0 &pwr_irq 5 IRQ_TYPE_EDGE_FALLING 0>,
-					<61 0 &intc GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH>,
-					<65 0 &intc GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>,
-					<68 0 &intc GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>,
-					<69 0 &intc GIC_SPI 94  IRQ_TYPE_LEVEL_HIGH>,
-					<70 0 &intc GIC_SPI 62  IRQ_TYPE_LEVEL_HIGH>,
-					<73 0 &intc GIC_SPI 129 IRQ_TYPE_LEVEL_HIGH>;
-			};
+			interrupts-extended =
+				<&intc GIC_SPI 6   IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_0 */
+				<&intc GIC_SPI 7   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 8   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 9   IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 10  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 23  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 64  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 65  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 66  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 67  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 40  IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_10 */
+				<&intc GIC_SPI 42  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 76  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 77  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 121 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 1   IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 3   IRQ_TYPE_LEVEL_HIGH>,
+				<0>,						/* EXTI_20 */
+				<&intc GIC_SPI 31  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 33  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 72  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 95  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 107 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 37  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 38  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 39  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 71  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 52  IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_30 */
+				<&intc GIC_SPI 53  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 82  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 83  IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,						/* EXTI_40 */
+				<0>,
+				<0>,
+				<&intc GIC_SPI 75  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 98  IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 93  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 138 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<&intc GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_50 */
+				<0>,
+				<&intc GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 141 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 135 IRQ_TYPE_LEVEL_HIGH>,
+				<&pwr_irq 0 IRQ_TYPE_EDGE_FALLING 0>,
+				<&pwr_irq 1 IRQ_TYPE_EDGE_FALLING 0>,
+				<&pwr_irq 2 IRQ_TYPE_EDGE_FALLING 0>,
+				<&pwr_irq 3 IRQ_TYPE_EDGE_FALLING 0>,
+				<&pwr_irq 4 IRQ_TYPE_EDGE_FALLING 0>,
+				<&pwr_irq 5 IRQ_TYPE_EDGE_FALLING 0>,		/* EXTI_60 */
+				<&intc GIC_SPI 100 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>,
+				<0>,
+				<0>,
+				<&intc GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 94  IRQ_TYPE_LEVEL_HIGH>,
+				<&intc GIC_SPI 62  IRQ_TYPE_LEVEL_HIGH>,	/* EXTI_70 */
+				<0>,
+				<0>,
+				<&intc GIC_SPI 129 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
 		syscfg: syscon@50020000 {
-- 
2.39.2

